functions.inc.php

```diff
@@ -1918,4 +1918,21 @@ function getRemoteAddr() {
     return $REMOTE_ADDR;
 }
 
+
+/**
+ * Returns a hash for a username valid for one day
+ *
+ * @param String $username user name
+ * @return String password recovery code
+ */
+function getPasswordRecoveryCode($username)
+{
+    $username = trim(strtolower($username));
+    $date = date('Y-m-d');
+
+    $code = substr(strtoupper(md5('SECRET SALTING PHRASE' . $username . $date)), 0, 6);
```

Comment:
Why not use e.g. a truncation of a hash of the $CONF['setup_password'] - that'd give you something random to add into the hash creatio (and unique to this installation) to solve the problem?